core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -482,6 +492,9 @@ private Metadata fetchMetadata() {
                     + "instance.",
                 connectionName));
       }
+      
+      checkDatabaseCompatibility(instanceMetadata, enableIamAuth, connectionName);
```

Comment:
nit: just inlining this method would match the checks above.

Reply:
I did it this way because it's easier to unit test. I think it'd actually be better if we pulled out the other checks and moved all the checks into a utils class, then tested that. It would increase our unit test coverage